diesel/src/pg/expression/functions.rs

```diff
@@ -748,3 +748,75 @@ define_sql_function! {
+define_sql_function! {
+    /// Prepends an element to the beginning of an array
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_prepend;
+    /// #     use diesel::sql_types::{Nullable, Integer, Array};
+    /// #     let connection = &mut establish_connection();
+    /// let ints = diesel::select(array_prepend::<Integer, Array<_>, _, _>(3, vec![1, 2]))
+    ///     .get_result::<Vec<i32>>(connection)?;
+    /// assert_eq!(vec![3, 1, 2], ints);
+    ///
+    /// let ints = diesel::select(array_prepend::<Nullable<Integer>, Array<_>, _, _>(None::<i32>, vec![Some(1), Some(2)]))
+    ///     .get_result::<Vec<Option<i32>>>(connection)?;
+    /// assert_eq!(vec![None, Some(1), Some(2)], ints);
+    ///
+    /// let ints = diesel::select(array_prepend::<Integer, Nullable<Array<_>>, _, _>(3, None::<Vec<i32>>))
+    ///     .get_result::<Vec<i32>>(connection)?;
+    /// assert_eq!(vec![3], ints);
+    ///
+    /// let ints = diesel::select(array_prepend::<Nullable<Integer>, Nullable<Array<_>>, _, _>(None::<i32>, None::<Vec<i32>>))
+    ///     .get_result::<Vec<Option<i32>>>(connection)?;
+    /// assert_eq!(vec![None], ints);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn array_prepend<T: SingleValue, Arr: ArrayOrNullableArray<Inner=T> + SingleValue>(e: T, a: Arr) -> Array<T>;
+}
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Removes all elements equal to the given value from the array
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_remove;
+    /// #     use diesel::sql_types::{Nullable, Integer, Array};
+    /// #     let connection = &mut establish_connection();
+    /// let ints = diesel::select(array_remove::<Array<_>, Integer, _, _>(vec![1, 2, 3, 2], 2))
+    ///     .get_result::<Option<Vec<i32>>>(connection)?;
+    /// assert_eq!(Some(vec![1, 3]), ints);
+
+    /// let ints = diesel::select(array_remove::<Array<_>, Nullable<Integer>, _, _>(vec![None, Some(1), Some(2), None, Some(4)], None::<i32>))
+    ///     .get_result::<Option<Vec<Option<i32>>>>(connection)?;
+    /// assert_eq!(Some(vec![Some(1), Some(2), Some(4)]), ints);
+
+    /// let ints = diesel::select(array_remove::<Nullable<Array<_>>, Nullable<Integer>, _, _>(None::<Vec<i32>>, None::<i32>))
+    ///     .get_result::<Option<Vec<Option<i32>>>>(connection)?;
+    /// assert_eq!(None, ints);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn array_remove<Arr: ArrayOrNullableArray<Inner=T> + SingleValue, T: SingleValue>(a: Arr, e: T) -> Nullable<Array<T>>;
```

Comment:
I think most of the time we would use this function with a not null array, so it would be nice if we can remove the `Nullable<_>` from the return time when isn't need.  
```suggestion
    fn array_remove<Arr: ArrayOrNullableArray<Inner=T> + SingleValue, T: SingleValue>(a: Arr, e: T) -> Arr;
```

Reply:
Double checking the behaviour on postgres:
- `SELECT array_remove(ARRAY[1, 2, 3], NULL);` returns `ARRAY[1, 2, 3]`

Reply:
I start the review before [your explanation](https://github.com/diesel-rs/diesel/pull/4187#discussion_r1723833738) :sweat_smile:, but I think it answers it